internal/database/types_hcpcluster.go

```diff
@@ -0,0 +1,139 @@
+package database
+
+import (
+	"strings"
+	"encoding/json"
+
+	"github.com/google/uuid"
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	azcorearm "github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+)
+
+type BaseDocument struct {
+	ID string `json:"id,omitempty"`
+
+	// https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/time-to-live
+	TimeToLive int `json:"ttl,omitempty"`
+
+	// System-defined properties generated by Cosmos DB
+	CosmosResourceID  string      `json:"_rid,omitempty"`
+	CosmosSelf        string      `json:"_self,omitempty"`
+	CosmosETag        azcore.ETag `json:"_etag,omitempty"`
+	CosmosAttachments string      `json:"_attachments,omitempty"`
+	CosmosTimestamp   int         `json:"_ts,omitempty"`
+}
+
+// newBaseDocument returns a baseDocument with a unique ID.
+func NewBaseDocument() BaseDocument {
+	return BaseDocument{ID: uuid.New().String()}
+}
+
+type TypedDocument struct {
+	BaseDocument
+	PartitionKey string          `json:"partitionKey"`
+	ResourceType string          `json:"resourceType"`
+	Properties   json.RawMessage `json:"properties"`
+}
+
+// newTypedDocument returns a TypedDocument from a ResourceType.
+func NewTypedDocument(partitionKey string, resourceType azcorearm.ResourceType) *TypedDocument {
+	return &TypedDocument{
+		BaseDocument: NewBaseDocument(),
+		PartitionKey: strings.ToLower(partitionKey),
+		ResourceType: strings.ToLower(resourceType.String()),
+	}
+}
```

Comment:
These only exist so the diff didn't get ridiculous by exposing the types, but it might have been clearer to do so.  At any rate, I think these should be exposed for consumers to inspect, but NOT persist.  Notice below persistence is disallowed.